Documentation/Yarn-Spec.md

```diff
@@ -0,0 +1,636 @@
+
+There must only be a single else command (if any) per flow control.
+The else command must go after the if and any elseif commands.
+The else must not exist without an if command.
+
+The example of the else command is `<<else>>`.
+
+#### endif
+
+The _endif_ command is the final element of flow control and is comprised solely of the keyword `endif`.
+The endif must be present whenever there is flow control and must go after the if and any elseif or else commands.
+The endif exists to allow the implementing program know when the scope of the other elements in the flow control has ended.
+`<<endif>>` is the example of the endif command.
+
+#### Scope and Blocks
+
+For the flow control to be useful there needs to be yarn statements which are run only when their appropriate expression evaluates to true.
+Flow control allows for blocks of statements to be scoped to their commands.
+A _block_ is a collection of statements that are scoped to a particular part of the flow control.
+
+The _scope_ of a block is determined by the flow control commands and associates each block with a command.
+The if, elseif, and else commands all have a block associated with them.
+The block of statements for a command start from the first statement after the command up until the next command in the flow control.
+When dealing with nested flow control the deepest set of flow control commands are to be the ones that can first assume another command closes their scope.
+
+While it is common for writers to indent their blocks relative to their scope it must not be used by the implementing program to determine scope.
+
+#### Handling 
+
+The implementing program must process all statements within the active blocks scope.
+The _active block_ is the block of yarn who's command expression evaluates to `true`.
+The block associated with the else command, if present, must only be determined as the active block if all other blocks expressions evaluate to false.
+
+An implementing program must not process any statements inside a block that is not the active block.
+An implementing program must only have, at most, one active block.
+If no blocks expression evaluates to true then no block must be processed.
+
+#### Ambiguity
+
+Because the flow control commands allow for potentially multiple commands and their blocks to be the true one, the implementing program must select them in a top down approach when there is conflicting flow.
+For example take the following flow control:
+
+```yarn
+<<if false>>
+    if-scope
+<<elseif true>>
+    elseif-1-scope
+<<elseif true>>
+    elseif-2-scope
+<<else>>
+    else-scope
+<<endif>>
+```
+
+Both of the elseif commands expressions evaluate to true, so either ones attached block could be run and seen to be correct.
+However because one is above the other the block with `elseif-1-scope` dialogue inside would be the selected one.
+The implementing program should attempt to identify these scenarios however and alert the writer.
```

Comment:
This is contradictory with line 377 and, as far as I know, has no precedent in any scripting language. Recommend this requirement is removed in its entirety (additionally, "alert" is vague)

Reply:
Could you elaborate on how this contradicts line 377 and how this is without precedent as that line says *"Because the flow control commands allow for potentially multiple commands and their blocks to be the true one, the implementing program must select them in a top down approach when there is conflicting flow"* which is the case in I think literally every programming language I have used (scripting or not).
Both of the else-if statements here evaluate to true, so either could be seen as the valid, so the first one is chosen.

As to alert I agree, not the best wording.
My thinking when writing this was how compilers throw warnings of unreachable code, as I have seen multiple times writers stuff up their logic and then later be confused why they can't get certain branches be reached and is most of the time trivially detectable.

Reply:
The issue is line 394, not the rest of it - obviously if-else chains work in the way outlined. I took "alert" here to mean "throw an error" which would mean line 377 "must select them in a top down approach" would no longer be true - it wouldn't select anything because operation has halted!

Regardless, requiring the implementing program to perform unreachable code checks is an overreach. The minimal implementing program exists to run the script, not act as a linter or syntax checker